engine/api/src/org/apache/cloudstack/engine/subsystem/api/storage/HypervisorHostListener.java

```diff
@@ -21,7 +21,13 @@
 import com.cloud.exception.StorageConflictException;
 
 public interface HypervisorHostListener {
+    boolean hostAdded(long hostId);
+
     boolean hostConnect(long hostId, long poolId) throws StorageConflictException;
 
     boolean hostDisconnected(long hostId, long poolId);
+
+    boolean hostAboutToBeRemoved(long hostId);
```

Comment:
I am assuming this event is generated when moving to `Removing` state. Is there any action that is performed here which cannot be done when the host is in `Removed` state?